fdbserver/VersionedBTree.actor.cpp

```diff
@@ -2152,19 +2196,25 @@ class DWALPager final : public IPager2 {
 	// If the file already exists, pageSize might be different than desiredPageSize
 	// Use pageCacheSizeBytes == 0 to use default from flow knobs
 	// If memoryOnly is true, the pager will exist only in memory and once the cache is full writes will fail.
+	// Note that ownership is not taken for keyProvider and it must outlive the pager
 	DWALPager(int desiredPageSize,
 	          int desiredExtentSize,
 	          std::string filename,
 	          int64_t pageCacheSizeBytes,
 	          int64_t remapCleanupWindowBytes,
 	          int concurrentExtentReads,
-	          bool memoryOnly = false,
+	          bool memoryOnly,
+	          std::shared_ptr<IEncryptionKeyProvider> keyProvider,
```

Comment:
Just simplicity because Reference<> offers no benefit here.  In general we should use standard types when we can. 

Reference<> is also more work to use here because this is a shared_ptr of an interface.  Reference<> is basically intrusive reference counting, it requires that the T inherit ReferenceCounted<T> so it includes the reference count.  This has lower overhead and better memory locality in most cases vs shared_ptr<T>, but it gets awkward for interfaces, where Reference<SomeInterface> requires each implementation to define virtual addref() and delref() methods and inherit Reference<Implementation>.

Reply:
shared_ptr has atomic operations which has higher overhead. Though if it only being copied rarely and pass by reference most of the time, it should be fine.

Reply:
looks like it is the case though it is copied rarely.